Allow the detection rail to detect item slide over it 让探测滑轨能检测滑过的掉落物

WhereisFff opened the pull request:
- resolved #2882

WhereisFff committed:
让探测滑轨能检测滑过的掉落物

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/block/sliding/DetectorSlidingRailBlock.java:
```diff
@@ -94,7 +95,8 @@ public VoxelShape getShape(
     @Override
     protected void tick(BlockState state, ServerLevel level, BlockPos pos, RandomSource random) {
-        if (state.getValue(POWERED) && level.getEntitiesOfClass(SlidingBlockEntity.class, new AABB(pos.above())).isEmpty()) {
+        if (state.getValue(POWERED) && level.getEntitiesOfClass(SlidingBlockEntity.class, new AABB(pos.above())).isEmpty() 
```
请在“(”之后，“)”之前，每个“&&”之前进行换行

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/block/sliding/BaseSlidingRailBlock.java:
```diff
@@ -103,4 +106,19 @@ private boolean canStickTo(
+    public void entityInside(BlockState state, Level level, BlockPos pos, Entity entity) {
+        if (level.isClientSide) return;
+        if (entity instanceof ItemEntity) {
+            AABB railBox = new AABB(pos.getX(), pos.getY(), pos.getZ(), 
```
请在“(”之后，“)”之前，每个“,”之后进行换行

anildazzlee commented:
escort service in bangalore kerala malayali mallu working women independent mallu tamil hanty kannada student available in bangalore please contact anil 9778018722

WhereisFff committed:
让探测滑轨能检测滑过的掉落物

WhereisFff committed:
修style

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/block/sliding/BaseSlidingRailBlock.java:
```diff
@@ -86,21 +89,33 @@ public boolean isStickyBlock(BlockState state) {
-               || pos.relative(Direction.Axis.Y, 1).equals(otherPos)
-               || pos.relative(Direction.Axis.Z, -1).equals(otherPos)
-               || pos.relative(Direction.Axis.Z, 1).equals(otherPos);
+        return pos.relative(Direction.Axis.X, -1).equals(otherPos) || pos.relative(Direction.Axis.X, 1).equals(otherPos) || pos.relative(
```
此处换行请于每个“||”之前

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/block/sliding/DetectorSlidingRailBlock.java:
```diff
@@ -60,12 +64,11 @@ public DetectorSlidingRailBlock(Properties properties) {
     public BlockState getStateForPlacement(BlockPlaceContext context) {
         Direction facing = context.getHorizontalDirection();
-        if (context.getPlayer() != null && context.getPlayer().isShiftKeyDown()) {
+        if (context.getPlayer() != null
```
请在“(”之后，“)”之前，每个“&&”之前进行换行

Gu-ZT replied:
请在“(”之后，“)”之前，每个“&&”之前进行换行

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/block/sliding/DetectorSlidingRailBlock.java:
```diff
@@ -94,7 +92,10 @@ public VoxelShape getShape(
     @Override
     protected void tick(BlockState state, ServerLevel level, BlockPos pos, RandomSource random) {
-        if (state.getValue(POWERED) && level.getEntitiesOfClass(SlidingBlockEntity.class, new AABB(pos.above())).isEmpty()) {
+        if (state.getValue(POWERED)
```
请在“(”之后，“)”之前，每个“&&”之前进行换行

WhereisFff committed:
修style

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/block/sliding/BaseSlidingRailBlock.java:
```diff
@@ -98,9 +98,26 @@ private boolean canStickTo(
+               || pos.relative(
+            Direction.Axis.Y,
+            -1
+        ).equals(otherPos) || pos.relative(Direction.Axis.Y, 1).equals(otherPos) || pos.relative(
```
此处换行请于每个“||”之前

Gu-ZT committed:
refactor(sliding): 优化滑轨方块代码结构

- 简化 BaseSlidingRailBlock 中的相邻位置判断逻辑
- 移除 DetectorSlidingRailBlock 中不必要的 NotNull 注解
- 优化 DetectorSlidingRailBlock 的条件判断格式
- 使用 Optional 明确处理可能为空的 BlockEntity
- 统一滑轨方块实体交互逻辑的代码风格
- 更新贡献者列表顺序和新增贡献者信息
